src/consensus/cost_calculator.py

```diff
@@ -19,13 +19,18 @@ class CostResult(Streamable):
 
 
 def calculate_cost_of_program(
-    program: SerializedProgram, clvm_cost_ratio_constant: int, strict_mode: bool = False
+    program: SerializedProgram,
+    generator_ref_list: Optional[SerializedProgram],
+    clvm_cost_ratio_constant: int,
+    strict_mode: bool,
 ) -> CostResult:
     """
     This function calculates the total cost of either a block or a spendbundle
     """
     total_clvm_cost = 0
-    error, npc_list, cost = get_name_puzzle_conditions(program, strict_mode)
+    # TODO: add cost for generator refs
+    gen_refs = NilSerializedProgram if generator_ref_list is None else generator_ref_list
```

Comment:
Not sure you need a special case here? You're making a list of `CLVMObject` objects, so wouldn't an empty list generalize nicely?